### ISSUE TITLE
Whitespace beside a comma of a function is read out of the comma, while the parser gives it to a neighbour: the fix writes nowhere, or beside the run instead of over it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#349]
### What minimal example or steps are needed to reproduce the bug?

`functionCommaSpaceFix` names the whitespace on one side of a comma by that comma's own `before` and `after`. `postcss-value-parser` gives a whitespace run to whichever neighbour will hold it, and for three positions that neighbour is not the comma:

```
f(a, ,c)   div ","  before ""   after " "     the run in front of the second comma is the first comma's `after`
           div ","  before ""   after ""

f( ,a)     fn  "f"  before " "  after ""      the run in front of the only comma is the function's `before`
           div ","  before ""   after ""

f(a, )     fn  "f"  before ""   after " "     the run behind the only comma is the function's `after`
           div ","  before ""   after ""
```

So the span the fix writes comes out empty and standing beside the text it meant to replace rather than over it. Two things follow, and both are visible with no comment and no custom syntax involved.

**A warning that survives every run of `--fix`**, wherever the option takes whitespace away:

| rule | option | code | reported | what `--fix` writes |
|---|---|---|---|---|
| `function-comma-space-before` | `"never"` | `a { b: f(a, ,c); }` | `1:13` | nothing, run after run |
| `function-comma-space-before` | `"never"` | `a { b: f(a, , ,c); }` | `1:13` and `1:15` | nothing, run after run |
| `function-comma-space-before` | `"never"` | `a { b: f( ,a); }` | `1:11` | nothing, run after run |
| `function-comma-space-after` | `"never"` | `a { b: f(a, ); }` | `1:11` | nothing, run after run |
| `function-comma-newline-before` | `"never-multi-line"` | `a { b: f(a,` ⏎ `,c); }` | `2:1` | nothing, run after run |
| `function-comma-newline-after` | `"never-multi-line"` | `a { b: f(a,` ⏎ `); }` | `1:11` | nothing, run after run |

`"never-single-line"` goes the same way as `"never"` on every single-line shape above.

**A file that grows a character on every run**, wherever the option writes whitespace in: the character lands at the comma, behind the run instead of over it, so the option is no nearer satisfied than it was.

| rule | option | code | first run | second run |
|---|---|---|---|---|
| `function-comma-space-before` | `"always"` | `a { b: f(a,` ⏎ `,c); }` | `a { b: f(a ,` ⏎ ` ,c); }` | `a { b: f(a ,` ⏎ `  ,c); }` |
| `function-comma-space-before` | `"always"` | `a { b: f(` ⏎ `,a); }` | `a { b: f(` ⏎ ` ,a); }` | `a { b: f(` ⏎ `  ,a); }` |
| `function-comma-space-after` | `"always"` | `a { b: f(a,` ⏎ `); }` | `a { b: f(a, ` ⏎ `); }` | `a { b: f(a,  ` ⏎ `); }` |

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-comma-space-before": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The whitespace standing on either side of a comma is written like any other: taken away under `never` and `never-multi-line`, replaced under `always`, and no warning left standing behind a run of `--fix`.

### What actually happened?

Nothing is written and the warning comes back on every run, or the character is written beside the whitespace it was meant to replace and the file grows on every run.

### Do you have a proposal to fix the bug?

Ask where that whitespace stands rather than assuming the comma holds it. The run in front of a comma is the previous divider's `after` where a divider stands in front, the function's `before` where the comma opens the arguments, and the comma's own `before` otherwise; the run behind it is the function's `after` where the comma closes them, and the comma's own `after` otherwise. The fix is handed the comma's place among the arguments already, so it needs the function node in place of the bare list of them to answer this.

Whitespace between the opening parenthesis and a comma is written by `function-parentheses-space-inside` as well, so a configuration asking one of the two rules for a space there and the other for none cannot be satisfied by any file — the same standing between `value-list-comma-space-before` and `declaration-colon-space-after` since [#166](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/166).

Found while fixing [#275](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/275), and reproduced on `main` before that branch as well.
